modules/call-dispatch/src/lib.rs

```diff
@@ -289,6 +311,25 @@ mod tests {
 
 	type MessageId = [u8; 4];
 
+	// These derives are a bit gross but we need all of them to ensure we fit into Substrate's
+	// definition of what an `AccountId` is
+	#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq, Default, PartialOrd, Ord, Serialize, Deserialize)]
+	pub struct TestAccountId(AccountId);
```

Comment:
This doesn't seem right, since we have `TestAccountPublic`, which identifies `AcocuntIds` we should simply be able to use `AccountId` - this extra `IdentifyAccount` shouldn't be necessary.

Reply:
If you're referring to the typedef for `AccountId`, that's not going to work. However, I didn't try using `TestRuntime::AccountId` - that might be a better approach

Reply:
This was required since we were using the wrong type in `verify_message_origin()`